src/pkg/task/task.go

```diff
@@ -37,3 +51,168 @@ type Manager interface {
 	// Get the log of the specified task
 	GetLog(ctx context.Context, id int64) (log []byte, err error)
 }
+
+// NewManager creates an instance of the default task manager
+func NewManager() Manager {
+	return &manager{
+		dao:      dao.NewTaskDAO(),
+		jsClient: cjob.GlobalClient,
+		coreURL:  config.GetCoreURL(),
+	}
+}
+
+type manager struct {
+	dao      dao.TaskDAO
+	jsClient cjob.Client
+	coreURL  string
+}
+
+func (m *manager) Create(ctx context.Context, executionID int64, jb *Job, extraAttrs ...map[string]interface{}) (int64, error) {
+	// create task record in database
+	id, err := m.createTaskRecord(ctx, executionID, extraAttrs...)
+	if err != nil {
+		return 0, err
+	}
+	log.Debugf("the database record for task %d created", id)
+
+	// submit job to jobservice
+	m.submitJob(ctx, id, jb)
+
+	return id, nil
+}
+
+func (m *manager) createTaskRecord(ctx context.Context, executionID int64, extraAttrs ...map[string]interface{}) (int64, error) {
+	extras := map[string]interface{}{}
+	if len(extraAttrs) > 0 && extraAttrs[0] != nil {
+		extras = extraAttrs[0]
+	}
+	data, err := json.Marshal(extras)
+	if err != nil {
+		return 0, err
+	}
+
+	now := time.Now()
+	return m.dao.Create(ctx, &dao.Task{
+		ExecutionID:  executionID,
+		Status:       job.PendingStatus.String(),
+		StatusCode:   job.PendingStatus.Code(),
+		ExtraAttrs:   string(data),
+		CreationTime: now,
+		UpdateTime:   now,
+	})
+}
+
+func (m *manager) submitJob(ctx context.Context, id int64, jb *Job) {
```

Comment:
Maybe `submitJob` returns an error and status updating can be put into the `Create` method, which will be more clear.